src/Microsoft.Identity.Web/Policy/ScopeAuthorizationHandler.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Identity.Web
+{
+    /// <summary>
+    ///  Scope authorization handler that needs to be called for a specific requirement type.
+    ///  In this case, <see cref="ScopeAuthorizationRequirement"/>.
+    /// </summary>
+    internal class ScopeAuthorizationHandler : AuthorizationHandler<ScopeAuthorizationRequirement>
+    {
+        private readonly IConfiguration _configuration;
+
+        /// <summary>
+        /// Constructor for the scope authorization handler, which takes a configuration.
+        /// </summary>
+        /// <param name="configuration">Configuration.</param>
+        public ScopeAuthorizationHandler(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
+        /// <summary>
+        ///  Makes a decision if authorization is allowed based on a specific requirement.
+        /// </summary>
+        /// <param name="context">AuthorizationHandlerContext.</param>
+        /// <param name="requirement">Scope requirement.</param>
+        /// <returns>Task.</returns>
+        protected override Task HandleRequirementAsync(
+            AuthorizationHandlerContext context,
+            ScopeAuthorizationRequirement requirement)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            // The resource is either the HttpContext or the Endpoint directly when used with the
+            // authorization middleware
+            var endpoint = context.Resource switch
+            {
+                HttpContext httpContext => httpContext.GetEndpoint(),
+                Endpoint ep => ep,
+                _ => null,
+            };
+
+            var data = endpoint?.Metadata.GetMetadata<IAuthRequiredScopeMetadata>();
+
+            IEnumerable<string>? scopes = null;
+            if (requirement?.RequiredScopesConfigurationKey != null)
+            {
+                scopes = _configuration.GetValue<string>(requirement?.RequiredScopesConfigurationKey)?.Split(' ');
+            }
+
+            if (scopes == null)
+            {
+                scopes = requirement?.AllowedValues ?? data?.AcceptedScope;
+            }
+
+            // Can't determine what to do without scope metadata, so proceed
+            if (scopes is null)
+            {
+                //context.Succeed(requirement);
```

Comment:
Don't you want this to succeed since it's always on?

Reply:
When testing, it would proceed to the web API, but we don't want it to do that if the scopes are null. I'll need to take a look at this specific case again. thanks for pointing it out.

Reply:
@davidfowl So I get here when I have `[Authorize]` on the controller, but no `[RequiredScope()]` and have not defined a policy in the startup.cs. So I assume this means, we have to have an authenticated user, but no scope check? So succeed? Is that correct? What if there were another requirement in Startup.cs would the whole authorization succeed?

Reply:
> What if there were another requirement in Startup.cs would the whole authorization succeed?

There's no way to bypass every requirement being succeeded, although any handler COULD do that if they wanted to just iterate over all requirements and call Succeed on each, but this is just succeeding your requirement here.

Reply:
To David's point, if you just fall thru and don't call Succeed on this requirement, that will have the effect of blocking all requests without scope metadata

Reply:
got it, thanks @HaoK